src/library_manager/library_manager.c

```diff
@@ -0,0 +1,458 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Jaroslaw Stelter <jaroslaw.stelter@intel.com>
+//         Pawel Dobrowolski<pawelx.dobrowolski@intel.com>
+
+/*
+ * Dynamic module loading functions.
+ */
+
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component_ext.h>
+#include <sof/common.h>
+#include <sof/ipc/topology.h>
+#include <sof/lib/wait.h>
+
+#include <sof/sof.h>
+#include <sof/spinlock.h>
+#include <sof/library_manager.h>
+#include <sof/audio/module_adapter/module/generic.h>
+
+#ifdef LIBRARY_MANAGER
+#include <sof/audio/module_adapter/module/intel_modules.h>
+#endif
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+
+/* 54cf5598-8b29-11ec-a8a3-0242ac120002 */
+
+DECLARE_SOF_UUID("lib_manager", lib_manager_uuid, 0x54cf5598, 0x8b29, 0x11ec,
+		 0xa8, 0xa3, 0x02, 0x42, 0xac, 0x12, 0x00, 0x02);
+
+DECLARE_TR_CTX(lib_manager_tr, SOF_UUID(lib_manager_uuid), LOG_LEVEL_INFO);
+
+
+extern void *SystemAgentStart(uint32_t entry_point, uint32_t module_id,
+		uint32_t instance_id, uint32_t core_id, void *mod_cfg);
+
+
+/* TODO: Manifest size should be defined somewhere - to check!!!*/
```

Comment:
See comment at the top of `src/include/kernel/ext_manifest.h`
```
/*
 * Parts of this file are copied to ext_manifest.h in Linux kernel tree
 * and to the rimage tree. This duplication is OK because it is part of
 * a versioned ABI.
 */
```